### PR TITLE
AzureBicepCosmosDB -> AzureCosmosDB

### PR DESCRIPTION
**Pull request by mitchdenny**
- [x] remove old provisioner based logic.
- [x] remove old Cosmos DB types
- [x] rename bicep-based types
- [x] Make emulator improvements work with Bicep.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2352)

### CODE REVIEW

File: src/Aspire.Hosting.Azure/AzureCosmosDBResource.cs [hunk 1]
```diff
@@ -1,59 +1,171 @@
...
             .OfType<AllocatedEndpointAnnotation>()
             .FirstOrDefault(x => x.Name == endpointName)
             ?.Port
         ?? throw new DistributedApplicationException($"Azure Cosmos DB resource does not have endpoint annotation with name '{endpointName}'.");
+}
+
+/// <summary>
+/// A resource that represents an Azure Cosmos DB database.
+/// </summary>
+public class AzureCosmosDBDatabaseResource(string name, AzureCosmosDBResource cosmosDB) :
```
Comment (davidfowl) [inline src/Aspire.Hosting.Azure/AzureCosmosDBResource.cs:61]: I think we should nuke this.

File: src/Aspire.Hosting.Azure/AzureCosmosDBResource.cs [hunk 2]
```diff
@@ -1,59 +1,171 @@
...
+    }
+
+    /// <summary>
+    /// Adds a resource which represents a database in the associated Cosmos DB account resource.
+    /// </summary>
+    /// <param name="builder">AzureCosmosDB resource builder.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="databaseName">Name of database.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<AzureCosmosDBDatabaseResource> AddDatabase(this IResourceBuilder<AzureCosmosDBResource> builder, string name, string? databaseName = null)
```
Comment (davidfowl) [inline src/Aspire.Hosting.Azure/AzureCosmosDBResource.cs:159]: Keep this method but return AzureCosmosDBResource instead.

Comment (mitchdenny) [reply src/Aspire.Hosting.Azure/AzureCosmosDBResource.cs]: Done.

File: src/Aspire.Hosting.Azure/AzureCosmosDBResource.cs (hunk 1 above)
Comment (mitchdenny) [reply src/Aspire.Hosting.Azure/AzureCosmosDBResource.cs:61]: Done.

Comment (davidfowl) [reply src/Aspire.Hosting.Azure/AzureCosmosDBResource.cs:61]: Are you going to delete this class?

**Comment by davidfowl** [comment]
You forgot to delete the CosmosDbDatabaseResource.

**Review by davidfowl — APPROVED** [review]
Pending deletion of AzureCosmosDBDatabaseResource

**Pull request merged**